src/controllers/reactive-query-controller.ts

```diff
@@ -0,0 +1,108 @@
+import { type ReactiveController, type ReactiveControllerHost } from 'lit';
+import { liveQuery, type Subscription } from 'dexie';
+
+export interface ReactiveQueryOptions<T> {
+  query: () => Promise<T>;
+  initialValue?: T;
```

Comment:
Providing an initial value complicates the calling code unnecessarily. Just use undefined as the initial value internally. Externally, expose a method that models the different states of the subscription, similar to Lit's `Task.render()` method. The code that uses the query doesn't really care about an initial value. All it cares about is the loading state, stream of updated values once loading is done, and the error state if something goes wrong. Would a method similar to `Task.render()` be sufficient to cover all of the use cases of the query controller in this PR?